app/routes/wiki-page.js

```diff
@@ -134,9 +134,10 @@ export default Route.extend(
         const surrogateKeys = model.get('surrogateKeys');
         let redirectTo = model.get('redirectTo');
 
-        if (model.isRandomPage) {
+        if (model.redirected || model.isRandomPage) {
           this.transitionTo('wiki-page', encodeURIComponent(normalizeToUnderscore(model.title)));
```

Comment:
will `transitionTo` trigger another request to MW?

Reply:
it will respond with 30X response and the browser will make another request.